src/librustc_trans/type_of.rs

```diff
@@ -359,7 +352,7 @@ impl<'tcx> LayoutLlvmExt<'tcx> for TyLayout<'tcx> {
         };
         let scalar = [a, b][index];
 
-        // Make sure to return the same type `immediate_llvm_type` would,
+        // Make sure to return the same type `llvm_type` would,
```

Comment:
Hmm, looking at this again, the special-case for `bool` could go better in `scalar_llvm_type_at`, which would let you remove this whole comment too.